src/main/java/org/threadly/concurrent/future/SettableListenableFuture.java

```diff
@@ -187,6 +241,8 @@ public T get() throws InterruptedException, ExecutionException {
       
       if (failure != null) {
         throw new ExecutionException(failure);
```

Comment:
CancellationException follows the java.util.concurrent.Future contract for these .get() calls, I think what you described would be too confusing/unexpected.